src/node_crypto.cc

```diff
@@ -3466,6 +3465,9 @@ BaseObjectPtr<BaseObject> NativeKeyObject::KeyObjectTransferData::Deserialize(
 
   Local<Value> handle = KeyObjectHandle::Create(env, data_).ToLocalChecked();
   Local<Function> key_ctor;
+  Local<Value> arg = FIXED_ONE_BYTE_STRING(env->isolate(),
+                                              "internal/crypto/keys");
+  env->native_module_require()->Call(context, Null(env->isolate()), 1, &arg);
```

Comment:
This call here could fail to populate `env->crypto_key_object_secret_constructor()` etc., mostly because the existence of `worker.terminate()` means that *any* call into JS can fail with an exception, regardless of whether the code itself throws something or not.

So I would recommend checking for an exception here, and returning if there is one:

```suggestion
  if (env->native_module_require()->Call(context, Null(env->isolate()), 1, &arg).IsEmpty())
    return {};
```

Reply:
Yes i will incorporate this check, how should i proceed with writing a test for these changes? jasnell said he added a check in his open draft pull request. Looks like as part of refactoring, node_crypto.cc file is removed over there.

Reply:
I think you could basically take the snipped from the issue, add a `require('../common');` as the first statement, and maybe a comment that links to the issue, and then just save that as a file in `test/parallel/`.

> Looks like as part of refactoring, node_crypto.cc file is removed over there.

I think that PR might require a rebase or two anyway. Also, this PR should probably land at least on v14.x and v12.x as well, unlike the other one, so the backporting process is easier overall if this PR lands on its own and we don’t worry as much about the other one. :)